canals/pipeline/pipeline.py

```diff
@@ -294,9 +294,9 @@ def _direct_connect(self, from_node: str, from_socket: OutputSocket, to_node: st
         Directly connect socket to socket. This method does not type-check the connections: use 'Pipeline.connect()'
         instead (which uses 'find_unambiguous_connection()' to validate types).
         """
-        # Make sure the receiving socket isn't already connected - sending sockets can be connected as many times as needed,
-        # so they don't need this check
-        if to_socket.sender:
+        # Make sure the receiving socket isn't already connected, unless it's variadic. Sending sockets can be
+        # connected as many times as needed, so they don't need this check
+        if to_socket.sender and not to_socket.is_variadic:
```

Comment:
`InputSocket.sender` defaults to an empty list, so if `to_socket.sender` is truthy it means a connection already happend